plugins/obs-ffmpeg/obs-ffmpeg-mpegts.c

```diff
@@ -1115,6 +1114,8 @@ void mpegts_write_packet(struct ffmpeg_output *stream,
 
 	pthread_mutex_lock(&stream->write_mutex);
 	da_push_back(stream->packets, &packet);
+	av_free(packet->data);
+	packet = NULL;
 	pthread_mutex_unlock(&stream->write_mutex);
```

Comment:
Is this safe? This pushes a pointer to the packet into `stream->packets` and then immediately frees the data pointer, leaving `packet->data` pointing to freed memory. Did you mean to set `packet->data = NULL` or use `av_freep`?

Reply:
av_freep indeed; i moved it after av_interleaved_write_frame too.

Reply:
`packet = NULL;` now seems redundant as it's a stack variable that's no longer referenced on this code path.

Reply:
indeed, thx a lot as usual @notr1ch !